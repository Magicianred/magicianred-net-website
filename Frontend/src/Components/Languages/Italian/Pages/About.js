import React, { Component } from 'react';
import axios from 'axios';
import Jumbotron from 'react-bootstrap/Jumbotron';

import Layout from './../Partials/Layout';

import Paragraph from '../../../../Models/Paragraph';

const publicUrl = process.env.PUBLIC_URL;

/**
 * Component for showing the About Page page.
 * 
 * @component
 * @example
 * return (
 *   <About />
 * )
 */
class About extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            paragraphs: [],
            hasError: false,
        };
    }        

    componentDidMount() {
        const paragraphs = [];
        this.serverRequest = 
            axios
                .get(`${publicUrl}/data/languages/english/pages/about.json`)
                .then((result) => {
                    result.data[0].paragraphs.map((item, index) => paragraphs.push(new Paragraph(index, item.text)));
                    this.setState({ paragraphs });
            });
    }

    // componentWillUnmount() {
    //     if (this.serverRequest) {
    //         this.serverRequest.abort();
    //     }
    // }

    render() {
        const { items, paragraphs, hasError } = this.state;
        return (
            <Layout>
                {hasError ? <h1>Something went wrong.</h1> : null}
                <div className="page">
                    <Jumbotron className="page about">
                        <h2>About Me page</h2>
                        <p>This is the about me page</p>
                        <hr />
                        <div>
                            <img align="right" style={{ margin: "10px" }} src="http://placebeard.it/g/200/200" width="200" height="200" />
                            {paragraphs.map(paragraph => paragraph.toDisplay())}
                        </div>
                    </Jumbotron>
                </div>
            </Layout>
        );
    }
}

export default About;