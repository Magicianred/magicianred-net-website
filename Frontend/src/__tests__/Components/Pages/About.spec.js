import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';

import AboutSectionMenu from "./../../../Components/Partials/AboutSectionMenu";
import About from "./../../../Components/Pages/About";

Enzyme.configure({ adapter: new Adapter() });
let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("Show the About Us page", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default params)", () => {
            const instance = create(<About />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    it("Show about page", () => {
        act(() => {
            render(<About />, container);
        });
        expect(container.querySelector("h2").textContent).toBe("About Us page");
        expect(container.querySelector("p").textContent).toBe("This is the about us page");
    });

    it("Show the about section menu", () => {
        const wrapper = shallow(<About />);
        expect(wrapper.contains(<AboutSectionMenu />)).toBe(true);
    });

    it("Show link to Who Are page", () => {
        render(<About />, container);
        expect(container.querySelectorAll("a[href='/about/whoare']")[0].textContent).toBe("Who Are");
    });

    it("Show link to Where Are page", () => {
        act(() => {
            render(<About />, container);
        });
        expect(container.querySelectorAll("a[href='/about/whereare']")[0].textContent).toBe("Where Are");
    });
});