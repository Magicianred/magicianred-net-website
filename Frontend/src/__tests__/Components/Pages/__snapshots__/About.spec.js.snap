// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show the About Us page Snapshots Matches the snapshot (with default params) 1`] = `
<div
  className="page"
>
  <div
    className="page about jumbotron"
  >
    <h2>
      About Us page
    </h2>
    <p>
      This is the about us page
    </p>
    <hr />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas ut est non mauris iaculis vulputate. Vivamus luctus sapien et nisl mattis consequat. Vestibulum nec varius sem. Curabitur aliquet sapien in diam efficitur euismod. Fusce molestie nibh vel suscipit eleifend. Suspendisse dignissim et lectus at tempus. Nam feugiat lorem ut pharetra porta. Duis hendrerit hendrerit metus, vel laoreet libero commodo sed. Curabitur nunc eros, venenatis et diam a, hendrerit volutpat neque. Aliquam tempus cursus efficitur. Proin consequat fringilla massa, vitae imperdiet nibh feugiat at. Nulla malesuada vestibulum dolor et euismod. Donec eu urna urna.
    </p>
    <p>
      Sed imperdiet nunc eget ante aliquam auctor. Aliquam rhoncus quam sed sodales imperdiet. Aliquam erat volutpat. Sed mattis luctus pretium. Suspendisse tempor, nunc quis consectetur aliquam, metus orci porttitor augue, nec malesuada ipsum mauris eget tortor. Donec ut eleifend lorem. Quisque sagittis fermentum suscipit. Sed quis efficitur ex.
    </p>
    <p>
      Sed eget neque ante. Aliquam euismod ac lectus quis tempor. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nam placerat consectetur felis ut fermentum. Ut a mauris vel erat porta congue. Sed ut fermentum nisl, ac maximus turpis. Nunc rutrum, ipsum vitae laoreet rutrum, purus mi fermentum leo, ut pulvinar velit urna id leo. Vivamus eu mi enim. Praesent ac ultrices mi, lacinia interdum dui. Morbi tempus vulputate lacus ac luctus. Aenean in arcu interdum, efficitur quam eu, suscipit felis.
    </p>
  </div>
  <header
    className="partial about menu"
    style={
      Object {
        "borderBottom": "solid thin black",
        "borderTop": "solid thin black",
      }
    }
  >
    <nav
      className="justify-content-center navbar navbar-expand navbar-tabs"
      fill="true"
    >
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/about"
            href="/about"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            About Us
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/about/whoare"
            href="/about/whoare"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Who Are
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/about/whereare"
            href="/about/whereare"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Where Are
          </a>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
